glib2: build error libelf.h: No such file or directory

Maintainer: @tripolar @luizluca
Environment: lantiq_xrx200 x86_64 x86_geode, APU3, master openwrt-19.06

Description:

I have update my build yesterday (master / openwrt-19.06 ) and did a fresh build. I do not know what change/commit has introduced the problem, but recently I get the following **error** on my own builds.

> gresource-tool.c:32:20: fatal error: libelf.h: No such file or directory
>  #include <libelf.h>

I have seen in the logs before update the following messages

> /home/bbworker/bbworker/owrt_dev_x86_64/System6/build/openwrt/logs/target/linux/install.txt:1:warning: Cannot use CONFIG_STACK_VALIDATION=y, please install libelf-dev, libelf-devel or elfutils-libelf-devel

maybe this is related.

As a workaround I have installed  on debian the package libelf-dev with the command
```aptitude install libelf-dev```

I think it is a bad idea to depend on packages from the hostbuild package system. in this case **libelf-dev**.

I think it would be better that the Openwrt-buildsystem build his own libelf-dev for the host. So we do not have a build depend on the buildsystem (Distribution).

Would it make sense to build libelf-dev for the build host as well https://github.com/openwrt/openwrt/tree/master/package/libs/elfutils So we do not have do install libelf-dev from the build distribuiton?

--- a/libs/glib2/Makefile
+++ b/libs/glib2/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=glib2
 PKG_VERSION:=2.58.3
-PKG_RELEASE:=2
+PKG_RELEASE:=3
 
 PKG_SOURCE:=glib-$(PKG_VERSION).tar.xz
 PKG_BUILD_DIR:=$(BUILD_DIR)/glib-$(PKG_VERSION)
@@ -47,6 +47,7 @@ endef
 TARGET_CFLAGS += -Wno-error=implicit-function-declaration
 
 HOST_CONFIGURE_ARGS += \
+	--disable-libelf \
 	--disable-selinux \
 	--with-libiconv=gnu \
 	--with-pcre=internal \

